src/rgw/rgw_rados.cc

```diff
@@ -2578,7 +2578,12 @@ int RGWPutObjProcessor_Aio::prepare(RGWRados *store, string *oid_rand)
 int RGWPutObjProcessor_Atomic::handle_data(bufferlist& bl, off_t ofs, void **phandle, rgw_raw_obj *pobj, bool *again)
 {
   *phandle = NULL;
-  uint64_t max_write_size = MIN(max_chunk_size, (uint64_t)next_part_ofs - data_ofs);
+  uint64_t max_write_size =
+    MIN(max_chunk_size, (uint64_t)next_part_ofs - data_ofs);
+
+  /* accept full-stride, aligned writes */
+  if (max_write_size == 0)
+    max_write_size = max_chunk_size;
```

Comment:
@yehudasa that's not how you designed the iteration, right?  so although that induces non-zero length term, it doesn't really feel better to me by itself?